### ISSUE TITLE
Upgrade Carbon to Spark 2.0

### ISSUE THREAD

**Issue by ravipesala** [HuaweiBigData/carbondata#586]
Spark 2.0 is complete over haul of design. So we need to have major changes in Carbondata spark integration layer. Following are the few major tasks identified in Carbon.
1. Unified the context and introduced new concept called `SparkSession`.So carbon data also need to migrate to it.
2. Spark 2.0 sql parser has been upgraded to standard antlr4 grammar. And unified hive sql parser, so hive queries are no longer parsed with different parser. So our extended carbon parser does not work.
3. Aggregation interfaces got completely changed in Spark 2.0. So carbon also need to update aggregate interfaces to keep the push down aggregation works.

**Comment by ravipesala** [comment]
@jackylk Regarding point 2. 
There is a fixed antlr grammar in catalyst package. So I don't think we can add new grammar to it. So better to standardize all our sql commands to catalyst/hive grammar. And limit our custom commands like `show loads` etc to be parsed in our new separate parser.
Please comment.

**Comment by jackylk** [comment]
@ravipesala yes, I think we should have separate parser for new command introduced in carbon-spark module.
And I think this is out of the scope of carbon.

**Comment by chenliang613** [comment]
Currently,Spark 2.0 is not very stable and have many issues, so in the coming 6 months, most of users would still use spark 1.6. 
My opinion : prior to focus on supporting Spark 1.6 integration currently.

**Comment by allwefantasy** [comment]
For now ,Spark  2.0 is just preview version ,and it would take a long time to be stable until  version 2.1. So most of users use spark 1.6 before Spark  2.1 coming .  In order to encourage more people to use CarbonData, prior to focus on supporting Spark 1.6 integration currently would be the best choice.

**Comment by tanbamboo** [comment]
Although Spark 2.0 have many great improvements, I prefer to have a stable working CarbonData release version supporting Spark 1.6 first.

**Comment by jackylk** [comment]
I agree, having spark 1.6 integration support would be a better choice for most body.

**Comment by Vimal-Das** [comment]
@ravipesala Can you also enumerate the changes required for migrating to Spark 1.6.
